### PR TITLE
F/leaderboard sort paginate 7

### LINKED ISSUE #7: Feature: User can sort and view more pages in the leaderboard
## User Story

As a user, I'd like to be able to explore ALL the pets that the site has to offer. On the leaderboards page, I'd like to be able to sort the pets by 'most points' or 'least points' so that I can show love for pets that haven't been discovered yet.

## Your job

- Investigate [react-table](https://react-table-v7.tanstack.com/) v7 (v8 documentation isn't quite complete) and discuss here whether it is a good fit for this task
- Tasks specific to this ticket will be added below as comments as they are discussed.

### PR DESCRIPTION
**Pull request by rupert-denton**
Created pagination
Created sorting for leaderboard
Added animal images to leaderboard

Wrote tests 
Refactored test

### CODE REVIEW

File: client/components/__tests__/PointsTable.test.jsx [hunk 1]
```diff
@@ -1,34 +1,78 @@
...
+    const beforeFirstPetName = cells[1].textContent
+    expect(beforeFirstPetName).toContain('Pet 50')
+    const nameColHeader = within(headerRow).getAllByRole('columnheader')[1]
+    await userEvent.click(nameColHeader)
+    const afterFirstRow = within(table).getAllByRole('row')[1]
+    const afterFirstPetName =
+      within(afterFirstRow).getAllByRole('cell')[1].textContent
+    expect(afterFirstPetName).toContain('Pet 1')
+  })
 })
```
Comment (joshuavial) [inline client/components/__tests__/PointsTable.test.jsx:87]: Excellent work, can you please add a test to ensure the previous page button works too

**Review by joshuavial — CHANGES_REQUESTED** [review]
Excellent work - just one more test for the previous button and we're good to merge

File: client/components/__tests__/PointsTable.test.jsx (hunk 1 above)
Comment (matruane) [reply client/components/__tests__/PointsTable.test.jsx:87]: I've added a previous button click to the existing test.

**Review by joshuavial — APPROVED** [review]
nice work

**Pull request merged**